pkg/netstack/tcp_forwarder.go

```diff
@@ -51,15 +52,24 @@ func Unmap4in6(addr netip.Addr) netip.Addr {
 	return v4addr
 }
 
-func tcpHandler(ctx context.Context, upstream network.Network) func(req *tcp.ForwarderRequest) {
+func tcpHandler(ctx context.Context, upstream network.Network, overridePort string) func(req *tcp.ForwarderRequest) {
 	return func(req *tcp.ForwarderRequest) {
 		reqDetails := req.ID()
 
 		srcAddrPort := netip.AddrPortFrom(addrFromNetstackIP(reqDetails.RemoteAddress), reqDetails.RemotePort)
-		dstAddrPort := netip.AddrPortFrom(
-			Unmap4in6(addrFromNetstackIP(reqDetails.LocalAddress)),
-			reqDetails.LocalPort,
-		)
+
+		// Extract IPv4 address from IPv6-mapped address
+		dstAddr := Unmap4in6(addrFromNetstackIP(reqDetails.LocalAddress))
+		dstPort := reqDetails.LocalPort
+
```

Comment:
```suggestion
		dstAddrPort := netip.AddrPortFrom(
			Unmap4in6(addrFromNetstackIP(reqDetails.LocalAddress)),
			reqDetails.LocalPort,
		)
		
		if overridePort != 0 {
		    logger.Debug("Overriding port", "original_port", dstAddrPort.Port, "ovveride_port", overridePort)
        }
```

These comments are not needed